Composer/packages/client/src/components/ToolBar/index.tsx

```diff
@@ -30,29 +54,25 @@ function itemList(action, index) {
 // support ActionButton or React Elements, the display order is array index.
 // action = {type:action/element, text, align, element, buttonProps: use
 // fabric-ui IButtonProps interface}
-export function ToolBar(props) {
+export function ToolBar(props: ToolbarProps) {
+  const { toolbarItems = [], currentDialog, onCreateDialogComplete, ...rest } = props;
   const {
-    toolbarItems,
-    actions,
-    projectId,
-    currentDialog,
-    openNewTriggerModal,
-    onCreateDialogComplete,
-    onboardingAddCoachMarkRef,
-    showSkillManifestModal,
-    openDeleteBotModal,
-    ...rest
-  } = props;
-  let left = [];
-  let right = [];
-  if (toolbarItems && toolbarItems.length > 0) {
-    left = toolbarItems.filter((item) => {
-      return item.align === 'left';
-    });
-    right = toolbarItems.filter((item) => {
-      return item.align === 'right';
-    });
+    actions: { openNewTriggerModal, onboardingAddCoachMarkRef, showSkillManifestModal, createDialogBegin, exportToZip },
+    state: { projectId },
+  } = useContext(StoreContext);
```

Comment:
Check out the `useStoreContext` helper hook. This makes stubbing the store context much easier in test.

Reply:
👍